Bottom: c826c5b31c7468cca57c5bbc0dcc27010a6c23b7
Top:    c74173964b3ff05dd3c820fe5a182a1f87d0c0b1
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-24 18:30:32 +0100

hid/gtk: Simplify ghid_draw_area_update()

If the rect passed is NULL, we can simply pass that on to
gdk_window_invalidate_rect() which will invalidate the whole area.


---

diff --git a/src/hid/gtk/gui-utils.c b/src/hid/gtk/gui-utils.c
index 5b85111..840e7a9 100644
--- a/src/hid/gtk/gui-utils.c
+++ b/src/hid/gtk/gui-utils.c
@@ -152,19 +152,7 @@ ghid_button_state (GdkModifierType * state)
 void
 ghid_draw_area_update (GHidPort * port, GdkRectangle * rect)
 {
-  GdkRectangle update_rect;
-
-  if (rect)
-    update_rect = *rect;
-  else
-    {
-      update_rect.x = 0;
-      update_rect.y = 0;
-      update_rect.width = port->width;
-      update_rect.height = port->height;
-    }
-  gdk_window_invalidate_rect (port->drawing_area->window, &update_rect,
-			      FALSE);
+  gdk_window_invalidate_rect (port->drawing_area->window, rect, FALSE);
 }
